kernel/Utilities/BasisFunctions2DH1ConformingTriangle.cpp

```diff
@@ -173,29 +173,38 @@ double BasisFunction2DInteriorTriangle::evalDeriv1(
                     LobattoPolynomialDerivative(polynomialOrder1_, x1));
 }
 
-Base::BasisFunctionSet* createDGBasisFunctionSet2DH1Triangle(
+std::vector<Base::BaseBasisFunction*> createDGBasisFunctions2DH1Triangle(
     std::size_t order) {
-    Base::BasisFunctionSet* result(new Base::BasisFunctionSet(order));
+    std::vector<Base::BaseBasisFunction*> result;
     if (order > 0) {
         for (std::size_t i = 0; i < 3; ++i) {
```

Comment:
what does the `3` relate to?

Reply:
can you turn it into a variable?

Reply:
Could, but to do that nicely I need to go through `Geometry::ReferenceTetrahedron::Instance().numberOfVertices()` or so. Which to me feels not like making it much nicer.